[Tan Juay Hee] iP

juayhee opened the pull request:
# Duke
- CLI support
- text based
- **super fast** to use

> Your mind is for having ideas, not holding them - [David Allen](https://dansilvestre.com/productivity-quotes)

## Features
1) Add 3 types of tasks
    - Todo tasks 
    - Deadline tasks
    - Event tasks
```
todo project
deadline homework /by 11pm
event go to the zoo /at next monday
```
2) View tasks easily
3) Mark them as complete
4) Delete tasks as needed

## Coming soon
- [x] Task searching
- [x] Date recognition
- [ ] GUI

Commit:
Add Gradle support

juayhee committed:
Duke.java: Added formatAnswer method for formatting Duke's outputs

juayhee committed:
Duke.java: Updated main(): Duke will now echo all commands (except exit on bye)

juayhee committed:
Duke.java: Fixed javadocs formatting

juayhee committed:
Command.java: Added Command class

juayhee committed:
* Abstracted commands - Duke.main() now only calls a CommandFactory to generate the right type of Command. Command is the abstract type for all possible commands to Duke. That command is then executed (polymorphism takes care of correct behaviour).
* Echo function refactored to use Commands.
* Exit function (bye) refactored to use Commands.

juayhee committed:
Formatting

juayhee committed:
TaskManager.java: Added TaskManager, to manage a todo list for the user.

juayhee committed:
Updated scope of abstract method in Command to public

juayhee committed:
Added AddTaskCommand and ListCommand

juayhee committed:
TaskManager.java: Changed taskList to use ArrayList instead of array type.

juayhee committed:
AddTaskCommand.java:
* Now echos the task added back to the user
* Includes formatting

juayhee committed:
ListCommand.java: Added formatted taskList printing.

juayhee committed:
Added support for marking tasks as done/undone

juayhee committed:
Revert "Added support for marking tasks as done/undone"

This reverts commit 2f8c2dc958b51d2627b8d4277aa756db6fcea7d5.

juayhee committed:
Revert "Revert "Added support for marking tasks as done/undone""

This reverts commit bfabfc4f9348a645a03d4da768209147fb3a5d31.

juayhee committed:
Added Mark and UnmarkCommand, for keeping track of completion status of tasks in taskList.

juayhee committed:
Formatting update
* Added more static methods for formatting in Duke.java
* Refactored many Command classes to now use these new methods for formatting

juayhee committed:
Mark/Unmark formatting update
* Mark/Unmark commands now provide feedback in the console

juayhee committed:
Task inheritance
* Updated Task.java to become a parent class of more specific subtpyes of classes.

juayhee committed:
Added 3 types of tasks
* DeadlineTask
* EventTask
* TodoTask

juayhee committed:
Task.java: Updated tto become general parent class for the different types of tasks

juayhee committed:
CommandFactory.java: Added factory methods for 3 new tasks (Event, Deadline, Todo)

juayhee committed:
AddTaskCommand.java: Now adds the appropriate task to the TaskManager instead of just adding a Todo type task.

juayhee committed:
DeadlineTask: Formatting

juayhee committed:
DeadlineTask.java: Comment update

juayhee committed:
EventTask.java: Completed functionality

juayhee committed:
Duke lines
*Changed Duke's introduction and goodbye message

juayhee committed:
Semi-automated testing
*Updated expected and input files for use in testing

juayhee committed:
Exception handling:
*Added custom DukeException exception
*Handles errors for invalid commands
*Handles errors for commands with insufficient parameters

juayhee committed:
Added delete functionality for deleting tasks from the task list

juayhee committed:
test commit

test commit from mac

juayhee committed:
Delete README.md copy

test delete

juayhee committed:
Update EXPECTED.TXT

Update expected output for test

juayhee committed:
Fix Issue #1

juayhee committed:
Duke.java: Add tasklist.txt file on init functionality

Adds tasklist.txt file on program start if tasklist.txt does not yet exist. This is in preparation for saving task lists on the hard drive.

juayhee committed:
Update gitignore

Updated with tasklist.txt

Halpfrog reviewed (commented):
Well done with the naming of the variables. No issues with the coding standard.

Halpfrog commented on src/main/java/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+    public ArrayList<Task> taskList = TaskManager.taskList;
+    private int index; // Index of target task
+
+    public DeleteCommand(String i) {
```
Perhaps change this to deleteCommand(String i)

Halpfrog commented on src/main/java/ByeCommand.java:
```diff
@@ -3,7 +3,7 @@
 public class ByeCommand extends Command {
 
-    final String EXIT = "Bye. Hope to see you again soon!";
+    final String EXIT = "Duke terminated";
```
Good use of naming standards

juayhee committed:
Merge branch 'master' of https://github.com/juayhee/ip

juayhee committed:
Update Duke.java

Added part of new code to init tasklist file in user directory if it does not exist

juayhee committed:
Update Duke.java

Completed init tasklist file functionality

juayhee committed:
Update AddTaskCommand.java

Started writing functionality for writing to tasklist.txt

juayhee committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Duke.java

juayhee committed:
Update Task.java

Added getType command for retrieiving the type of task

juayhee committed:
Update AddTaskCommand.java

Now writes tasks to tasklist.txt. Uses the Task's .toString() format.

juayhee committed:
Refactor the way Duke updates tasklist.txt

* TaskManager now takes care of writing to tasklist.txt
* Commands now no longer add to the tasklist ARRAY directly. They do so through TaskManager.add()
* Added FileWriter close() to Duke.java after program finishes execution (cleanup)

juayhee committed:
Add task saving ability

* Duke can now save tasks in tasklist.txt
* TaskManager takes care of rewriting files when tasks get added or deleted

juayhee committed:
Added Duke functionality

* Duke can now load tasks from tasklist.txt
* Added new CustomTask class to manage adding tasks back into the tasklist array from the text file (should not be used otherwise)

juayhee committed:
Fix reader mistake in TaskManager.java

*Added reader.close() for BufferedReader at the end of loadTask

juayhee committed:
Change file writing process

*Duke now only writes to file at the end of execution

juayhee committed:
TaskManager.java: Delete unused code

juayhee committed:
Change DeadlineTask.java

* Deadlines now require yyyy-mm-dd formatted parameters, as the deadline date is now stored as LocalDate

juayhee committed:
Change date display

*Duke now displays date in d MM yyyy format.

juayhee committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/AddTaskCommand.java

juayhee committed:
Add Ui to manage user interactions

*Ui manages reading of inputs
*Ui manages printing of outputs

juayhee committed:
ByeCommand.java: Move system print to Ui

ByeCommand now relies on Ui to output to the user

juayhee committed:
ListCommand.java: Move system print to Ui

ListCommand now relies on Ui to output to the user (formatting of tasklist still done within ListCommand.java)

juayhee committed:
Ui.java: Improve output format of printAddTask()

juayhee committed:
MarkCommand.java: Move system print to ui

* MarkCommand now relies on Ui for outputs to user

juayhee committed:
MarkCommand.java: Deprecate unused code

juayhee committed:
Duke.java: Move introduction print to ui

* Duke now relies on Ui for outputs to user

juayhee committed:
DeleteCommand.java: Move delete confirmation print to ui

* DeleteCommand now relies on Ui for outputs to user

juayhee committed:
UnmarkCommand.java: Move delete confirmation print to ui

* UnmarkCommand now relies on Ui for outputs to user

juayhee committed:
Delete a bunch of deprecated code

juayhee committed:
Delete EchoCommand.java

* EchoCommand.java is completely unused

juayhee committed:
Duke.java: Delete deprecated formatting code

* All such funtionalities are now handled by Ui.java

juayhee committed:
Rename TaskManager to TaskList

juayhee committed:
Replace CommandFactory with Parser

juayhee committed:
Refactor code

* Storage now manages the loading and saving of tasks from tasklist.txt
* Tasklist.java now contains the tasklist array

tehkokhoe commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,68 @@
+    private LocalDate getDate() {
+        int index = this.findDeadlineIndex() + 4;
+        String date = this.input.substring(index);
+        LocalDate localDate = LocalDate.parse(date);
```
Should DateTimeParseException be handled?

tehkokhoe reviewed (commented):
I'm not sure how illegal date inputs are handled. Code is easy to read, and I couldn't find mistakes, good job! :)

viki0526 commented on src/main/java/AddTaskCommand.java:
```diff
@@ -0,0 +1,50 @@
+ * Generates the correct type of task and adds it to the task list
+ */
+
+public class AddTaskCommand extends Command {
+
+    private String input; // Details of the task
+    private String type; // Type of task
+    private boolean isComplete = false;
+
+    public AddTaskCommand(String input, String type) {
+        this.input = input;
+        this.type = type;
+    }
+
+    public AddTaskCommand(String input, String type, boolean isComplete) {
+        this.input = input;
+        this.type = type;
+        this.isComplete = isComplete;
+    }
```
I like your use of inheritance when structuring the Command classes as it makes for clean and logical code.

viki0526 commented on src/main/java/AddTaskCommand.java:
```diff
@@ -0,0 +1,50 @@
+ * Generates the correct type of task and adds it to the task list
+ */
+
+public class AddTaskCommand extends Command {
```
I believe class names should be nouns written in camelCasing according to coding standards.

viki0526 commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * Format DeadlineTask description into display format for Duke
+     */
+    private void updateDescription() {
+        this.description += this.getInfo();
+        this.description += "(by: ";
+        this.description += this.date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        this.description += ")";
```
I like your proper use of method naming standards(descriptive, camelCased verbs)

viki0526 reviewed (commented):
Well-structured classes for Commands and Tasks and very nice use of OOP principles.

juayhee committed:
Move all files into package: duke

juayhee committed:
Update gitignore

juayhee committed:
no message

juayhee committed:
no message

juayhee committed:
no message

juayhee committed:
no message

juayhee committed:
Duke.java: Fix error where duke tries to load the tasklist before there is any file

juayhee committed:
Add ParserTest to test the parser for Duke

juayhee committed:
Update ParserTest with more tests

juayhee committed:
Add TaskTest.java to test Task.java of Duke

juayhee committed:
no message

juayhee committed:
Add javadocs header comments

juayhee committed:
Formatting

juayhee committed:
Add search function for tasks

juayhee committed:
Merge branch 'branch-A-CodingStandard'

juayhee committed:
Merge branch 'branch-A-JavaDoc'

juayhee committed:
Fix additional whitespaces during printing of list

juayhee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

juayhee committed:
Gradle integration

juayhee committed:
Add GUI classes

juayhee committed:
Add GUI mockup

juayhee committed:
Add DialogBox stuff

juayhee committed:
Edit GUI to make Duke display text on one side and the user display text on another

juayhee committed:
Refactor userText variable

juayhee committed:
Link DukeGUI inputs to Duke

juayhee committed:
DukeGUI now prints output correctly

juayhee committed:
Fix output formatting

juayhee committed:
Update Gradle build file

juayhee committed:
Add assertions

juayhee committed:
Delete dead code

juayhee committed:
Merge pull request #3 from juayhee/branch-A-CodeQuality

Delete dead code

juayhee committed:
Merge branch 'master' into branch-A-Assertions

juayhee committed:
Merge pull request #4 from juayhee/branch-A-Assertions

Add assertions

juayhee committed:
Add command aliases to Duke

There is only one command for each action. It is inflexible for users who prefer command shorthands.

Add more aliases to each action.

This gives users more flexibility to choose what they want to use when inputting commands or event types.

Example: todo X is the same as t X

juayhee committed:
Add product website, fix bugs
